In [1]:
import pandas

In [8]:
my_zoo = pandas.read_csv("adopce-zvirat.csv", sep=";")
my_zoo.tail(35)

id                   nazev_cz                         nazev_en   
478  1379           Kočka palawanská              Palawan leopard cat  \
479  1381              Husa císařská                    Emperor goose   
480  1382             Včelojed lesní           European honey-buzzard   
481  1383  Kraska krátkoocasá jávská               Javan green magpie   
482  1384             Holub bělolící  Columba vitiensis griseogularis   
483  1385        Tkalčík červenozobý                                    
484  1386        Holub růžovotemenný                   Ducula rosacea   
485  1387              Husice rezavá               Tadorna ferruginea   
486  1391        Kachnička amazonská          Amazonetta brasiliensis   
487  1392         Chřástal kropenatý                  Porzana porzana   
488  1393             Lori sumbawský               Forsten's lorikeet   
489  1394       Stepokur hnědobřichý       Pterocles exustus Temminck   
490  1395                 Labuť malá               Cygnus columbianus   
491  1396             Kolpík africký                                    
492  1397            Pekari Wagnerův                Catagonus wagneri   
493  1398        Dikobraz palawanský                                    
494  1399              Kvakoš rezavý                                    
495  1401               Daman stepní                                    
496  1404              Dikdik Kirkův                                    
497  1405             Nosál bělohubý                                    
498  1406            Ibis šedokřídlý                                    
499  1407         Zoborožec luzonský                                    
500  1408                Bukač velký                                    
501  1409                Dytík bledý                                    
502  1411             Kačka strakatá                                    
503  1412            Želva hvězdnatá                                    
504  1413        Varanovec bornejský                                    
505  1414          Ropucha síťkovaná                                    
506  1415         Rosnička bělopruhá                                    
507  1416               Rosnice siná                                    
508  1417        Rosnice šalamounská                                    
509  1418             Létavka obecná                                    
510  1419                Bažant paví                                    
511  1420        Sojkovec rezavouchý                                    
512  1421         Kachnice kaštanová                                    

          trida_cz  cena  k_prohlidce  
478          Savci  5000            1  
479          Ptáci  2500            1  
480          Ptáci  3000            1  
481          Ptáci  3000            0  
482          Ptáci  1000            1  
483          Ptáci  1000            1  
484          Ptáci  1000            1  
485          Ptáci  1000            1  
486          Ptáci  1000            1  
487          Ptáci  1500            0  
488          Ptáci  1500            1  
489          Ptáci  2000            1  
490          Ptáci  2500            1  
491          Ptáci  3000            1  
492          Savci  2000            1  
493          Savci  2000            1  
494          Ptáci  2000            1  
495          Savci  2000            1  
496          Savci  3000            1  
497          Savci  5000            1  
498          Ptáci  3000            1  
499          Ptáci  2000            1  
500          Ptáci  2000            1  
501          Ptáci  1500            0  
502          Ptáci  1000            1  
503          Plazi  1500            1  
504          Plazi  5000            1  
505  Obojživelníci  1000            1  
506  Obojživelníci  1000            1  
507  Obojživelníci  1000            1  
508  Obojživelníci  1000            1  
509  Obojživelníci  1000            1  
510          Ptáci  1000            1  
511          Ptáci

In [3]:
my_zoo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   id           513 non-null    int64 
 1   nazev_cz     513 non-null    object
 2   nazev_en     513 non-null    object
 3   trida_cz     513 non-null    object
 4   cena         513 non-null    int64 
 5   k_prohlidce  513 non-null    int64 
dtypes: int64(3), object(3)
memory usage: 24.2+ KB


soubor má: 
512 řádků, 6 sloupců - id, název_cz, název_en, trida_cz.,  cena, k_prohlidce 

In [4]:
animal_number_34 = my_zoo.iloc[34]
animal_number_34[["nazev_cz","nazev_en"]]

nazev_cz     Ibis bílý
nazev_en    White ibis
Name: 34, dtype: object

Zvíře na řádku č. 34: 
nazev_cz     Ibis bílý
nazev_en    White ibis
Name: 34, dtype: object

Bonus: Využij metody sort_values() (a libovolných dalších metod), ke zjištění následujících informací:

    Adopce kterých zvířat je nejdražší?
    Které zvíře je alfabeticky poslední v češtině? Které v angličtině?

Bonus můžeš doplnit o nějaké vlastní pozorování.

In [5]:
animal_price = my_zoo.sort_values(by="cena")
animal_price[["nazev_cz","cena"]].tail(10)
                  

nazev_cz   cena
115                                Lev indický  30000
105                               Medvěd lední  30000
102                             Gorila nížinná  30000
244                             Varan komodský  30000
114                            Tygr sumaterský  50000
113                            Tygr ussurijský  50000
272                              Tygr malajský  50000
129                          Hroch obojživelný  50000
135  Žirafa severní núbijská (= Rothschildova)  50000
128                               Slon indický  50000

nejdražší zvířata k adopci jsou: 
114	Tygr sumaterský	50000
113	Tygr ussurijský	50000
272	Tygr malajský	50000
129	Hroch obojživelný	50000
135	Žirafa severní núbijská (= Rothschildova)	50000
128	Slon indický	50000

In [6]:
animal_name=my_zoo.sort_values(by="nazev_cz")
animal_name["nazev_cz"].tail(1)

371    Žluva hajní
Name: nazev_cz, dtype: object

In [10]:
animal_name_EN=my_zoo.sort_values(by="nazev_en")
animal_name_EN["nazev_en"].tail(1)

161    Yellow-necked francolin
Name: nazev_en, dtype: object

In [14]:
#průměrná cena  adopce 
price_ave = (my_zoo["cena"].mean()).round(0)
price_ave

3310.0